src/Symfony/Component/Serializer/Mapping/AttributeMetadataInterface.php

```diff
@@ -60,5 +60,5 @@ public function getMaxDepth();
     /**
      * Merges an {@see AttributeMetadataInterface} with in the current one.
      */
-    public function merge(self $attributeMetadata);
+    public function merge(AttributeMetadataInterface $attributeMetadata);
```

Comment:
the issue with this change is that PHP-CS-Fixer will revert it on any run.

Reply:
true, but we look at tests before merging, don't we? :)